src/github.com/matrix-org/dendrite/federationapi/routing/profile.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2017 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"net/http"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/accounts"
+	"github.com/matrix-org/dendrite/clientapi/httputil"
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/gomatrixserverlib"
+	"github.com/matrix-org/util"
+)
+
+type profileResponse struct {
+	AvatarURL   string `json:"avatar_url"`
+	DisplayName string `json:"displayname"`
+}
+
+type avatarURL struct {
+	AvatarURL string `json:"avatar_url"`
+}
+
+type displayName struct {
+	DisplayName string `json:"displayname"`
```

Comment:
These three structures look like duplication from https://github.com/matrix-org/dendrite/blob/master/src/github.com/matrix-org/dendrite/clientapi/routing/profile.go#L34-L45 we could easily avoid. I would have exported the ones in the client API (by capitilising the first letter of their name and updating references to them in `GetProfile`, `GetAvatarURL` and `GetDisplayName`, and also doc'ing them in comments otherwise linters will go crazy) to use them here instead.